File: DSA/Ayush_Mishra_590014844/Day_20_Question_2.c
#include <stdio.h>

void mergeArrays(int a[], int b[], int n, int m, int result[]) {
    int i = 0, j = 0, k = 0;
    while(i < n && j < m) {
        if(a[i] < b[j])
            result[k++] = a[i++];
        else
            result[k++] = b[j++];
    }
    while(i < n) result[k++] = a[i++];
    while(j < m) result[k++] = b[j++];
}

int main() {
    int n, m;
    printf("Enter size of first array: ");
    scanf("%d", &n);
    int a[n];
    printf("Enter elements of first sorted array: ");
    for(int i = 0; i < n; i++) scanf("%d", &a[i]);

    printf("Enter size of second array: ");
    scanf("%d", &m);
    int b[m];
    printf("Enter elements of second sorted array: ");
    for(int i = 0; i < m; i++) scanf("%d", &b[i]);

    int result[n + m];
    mergeArrays(a, b, n, m, result);

    printf("Merged array: ");
    for(int i = 0; i < n + m; i++) printf("%d ", result[i]);
    return 0;
}

File: DSA/Ayush_Mishra_590014844/Day_20_Question_2.py
def merge_sorted_arrays(arr1, arr2):
    merged = []
    i = j = 0
    while i < len(arr1) and j < len(arr2):
        if arr1[i] < arr2[j]:
            merged.append(arr1[i])
            i += 1
        else:
            merged.append(arr2[j])
            j += 1
    merged.extend(arr1[i:])
    merged.extend(arr2[j:])
    return merged

arr1 = list(map(int, input("Enter first sorted array: ").split()))
arr2 = list(map(int, input("Enter second sorted array: ").split()))
print("Merged array:", merge_sorted_arrays(arr1, arr2))
